Meta/run.py

```diff
@@ -0,0 +1,902 @@
+        config.add_devices(["sdhci-pci", "sd-card,drive=sd-boot-drive"])
+        config.kernel_cmdline.append("root=sd2:0:0")
+    elif config.usb_boot_enable:
+        config.boot_drive = f"if=none,id=usbstick,format=raw,file={config.disk_image}"
+        config.add_device("usb-storage,drive=usbstick")
+        # FIXME: Find a better way to address the usb drive
+        config.kernel_cmdline.append("root=block3:0")
+    else:
+        config.boot_drive = f"file={config.disk_image},format=raw,index=0,media=disk,id=disk"
+
+
+def determine_host_address() -> str:
+    return environ.get("SERENITY_HOST_IP", "127.0.0.1")
+
+
+def setup_gdb(config: Configuration):
+    config.enable_gdb = environ.get("SERENITY_DISABLE_GDB_SOCKET") != "1"
+    if config.qemu_kind == QEMUKind.NativeWindows or (
+        config.virtualization_support and config.qemu_kind == QEMUKind.MacOS
+    ):
+        config.enable_gdb = False
+
+    if config.enable_gdb:
+        config.extra_arguments.extend(["-gdb", f"tcp:{config.host_ip}:1234"])
+
+
+def setup_network_hardware(config: Configuration):
+    config.packet_logging_arguments = (environ.get("SERENITY_PACKET_LOGGING_ARG", "")).split()
+
+    provided_ethernet_device_type = environ.get("SERENITY_ETHERNET_DEVICE_TYPE")
+    if provided_ethernet_device_type is not None:
+        config.ethernet_device_type = provided_ethernet_device_type
+
+    if config.architecture == Arch.Aarch64:
+        config.network_backend = None
+        config.network_default_device = None
+    else:
+        config.network_backend = f"user,id=breh,hostfwd=tcp:{config.host_ip}:8888-10.0.2.15:8888,\
+hostfwd=tcp:{config.host_ip}:8823-10.0.2.15:23,\
+hostfwd=tcp:{config.host_ip}:8000-10.0.2.15:8000,\
+hostfwd=tcp:{config.host_ip}:2222-10.0.2.15:22"
+        config.network_default_device = f"{config.ethernet_device_type},netdev=breh"
+
+
+def setup_kernel(config: Configuration):
+    if config.architecture == Arch.Aarch64:
+        config.kernel_and_initrd_arguments = ["-kernel", "Kernel/Kernel"]
+    else:
+        config.kernel_and_initrd_arguments = ["-kernel", "Kernel/Prekernel/Prekernel", "-initrd", "Kernel/Kernel"]
+
+
+def setup_machine_devices(config: Configuration):
+    # TODO: Maybe disable SPICE everwhere except the default machine?
+
+    # Architecture specifics.
+    if config.architecture == Arch.Aarch64 and config.machine_type != MachineType.CI:
+        # FIXME: Windows QEMU crashes when we set the same display as usual here.
+        config.display_backend = None
+        config.display_device = None
+        config.audio_devices = []
+        config.extra_arguments.extend(["-serial", "stdio"])
+        config.qemu_machine = "raspi3b"
+        config.cpu_count = None
+        config.qemu_cpu = None
+        config.vga_type = None
+        return
+    else:
+        # Machine specific base setups (x86).
+        match config.machine_type:
+            case MachineType.QEMU35Grub | MachineType.QEMU35:
```

Comment:
I think this one should enable usb as well

Reply:
setting that flag here now.